commit/tokenprice/processor.go

```diff
@@ -47,6 +87,97 @@ func (w *Processor) ValidateObservation(
 	return nil
 }
 
+func (p *Processor) ObserveFDestChain() (*int, error) {
+	fChain, err := p.homeChain.GetFChain()
+	if err != nil {
+		// TODO: metrics
+		p.lggr.Warnw("call to GetFChain failed", "err", err)
+		return nil, nil
+	}
+
+	fDestChain, ok := fChain[p.cfg.DestChain]
+	if !ok {
+		return nil, fmt.Errorf("fChain does not have dest chain")
+	}
+
+	return &fDestChain, nil
+}
+
+func (p *Processor) ObserveFeedTokenPrices(ctx context.Context) []cciptypes.TokenPrice {
+	supportedChains, err := p.chainSupport.SupportedChains(p.oracleID)
+	if err != nil {
+		p.lggr.Warnw("call to SupportedChains failed", "err", err)
+		return []cciptypes.TokenPrice{}
+	}
+
+	if !supportedChains.Contains(cciptypes.ChainSelector(p.cfg.OffchainConfig.TokenPriceChainSelector)) {
+		p.lggr.Debugw("oracle does not support token price observation", "oracleID", p.oracleID)
+		return []cciptypes.TokenPrice{}
+
+	}
+
+	if p.tokenPriceReader == nil {
+		p.lggr.Errorw("no token price reader available")
+		return []cciptypes.TokenPrice{}
+	}
+
+	tokensToQuery := maps.Keys(p.cfg.OffchainConfig.TokenInfo)
+	p.lggr.Infow("observing feed token prices")
+	tokenPrices, err := p.tokenPriceReader.GetTokenFeedPricesUSD(ctx, tokensToQuery)
+	if err != nil {
+		p.lggr.Errorw("call to GetTokenFeedPricesUSD failed", "err", err)
+		return []cciptypes.TokenPrice{}
+	}
+
+	// Continue if we couldn't fetch all prices
+	if len(tokenPrices) != len(tokensToQuery) {
```

Comment:
The comment and the line after don't line up. What does "continue" mean in this context?